File: programmers/중복된 문자제거.py
def solution(my_string):
    my_list = list(my_string)
    for i in range(len(my_list)-1, -1, -1):
        if my_list.count(my_list[i]) >= 2:
            my_list.pop(i)
    return ''.join(i for i in my_list)









print(solution("people"))
print(solution("We are the world"))

File: programmers/코딩테스트 입문 with Java/중복된문자제거.java
package 코딩테스트입문;

import java.util.*;

/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/120888
 */
public class 중복된문자제거 {
    public static void main(String[] args) {
        System.out.println(solution("people"));
        System.out.println(solution("We are the world"));
    }
    public static String solution(String my_string) {
        Set<String> set = new HashSet<>();
        List<String> list =new ArrayList<>(Arrays.asList(my_string.split("")));
        StringBuilder sb= new StringBuilder();
        for (String s : list) {
            if(!set.contains(s)){
                set.add(s);
                sb.append(s);
            }
        }
        return sb.toString();
    }
}
